Error using mixed block size including 1D blocks

```julia
A = [1 2 1.; 0 0. 1; -2 1 4]
X0 = BallInf(ones(3), 0.1)
Σ = InitialValueProblem(LinearContinuousSystem(A), X0);

sol = solve(Σ, :T=>0.1, :vars=>[3], :partition=>[1:2, [3]], :apply_projection=>false,
                      :block_types=>Dict(HPolygon=>[1:2], LazySets.Interval=>[[3]]));

AssertionError: a linear map of size (2, 1) cannot be applied to a set of dimension 2

Stacktrace:
 [1] LazySets.LinearMap{Float64,Float64}(::Array{Float64,2}, ::LazySets.HPolygon{Float64}) at /Users/forets/.julia/v0.6/LazySets/src/LinearMap.jl:28
 [2] *(::Array{Float64,1}, ::LazySets.HPolygon{Float64}) at /Users/forets/.julia/v0.6/LazySets/src/LinearMap.jl:58
 [3] reach_blocks!(::Array{Float64,2}, ::Array{LazySets.LazySet{Float64},1}, ::Void, ::Reachability.ReachSets.##7#12{Float64}, ::Reachability.ReachSets.##10#15, ::Int64, ::Int64, ::Array{Int64,1}, ::Array{Union{Int64, UnitRange{Int64}},1}, ::Array{LazySets.CartesianProductArray{Float64,S} where S<:LazySets.LazySet{Float64},1}) at /Users/forets/.julia/v0.6/Reachability/src/ReachSets/reach_blocks.jl:147
 [4] #reach#6(::String, ::Float64, ::Type{LazySets.Hyperrectangle}, ::Float64, ::Type{LazySets.Hyperrectangle}, ::Bool, ::Bool, ::Type{T} where T, ::Bool, ::Array{Any,1}, ::Reachability.ReachSets.#reach, ::MathematicalSystems.InitialValueProblem{MathematicalSystems.LinearDiscreteSystem{Float64,Array{Float64,2}},LazySets.ConvexHull{Float64,LazySets.BallInf{Float64},LazySets.LinearMap{Float64,Float64}}}, ::Int64) at /Users/forets/.julia/v0.6/Reachability/src/ReachSets/reach.jl:162
 [5] (::Reachability.ReachSets.#kw##reach)(::Array{Any,1}, ::Reachability.ReachSets.#reach, ::MathematicalSystems.InitialValueProblem{MathematicalSystems.LinearDiscreteSystem{Float64,Array{Float64,2}},LazySets.ConvexHull{Float64,LazySets.BallInf{Float64},LazySets.LinearMap{Float64,Float64}}}, ::Int64) at ./<missing>:0
 [6] solve(::MathematicalSystems.InitialValueProblem{MathematicalSystems.LinearContinuousSystem{Float64,Array{Float64,2}},LazySets.BallInf{Float64}}, ::Reachability.Options) at /Users/forets/.julia/v0.6/Reachability/src/solve.jl:153
 [7] solve(::MathematicalSystems.InitialValueProblem{MathematicalSystems.LinearContinuousSystem{Float64,Array{Float64,2}},LazySets.BallInf{Float64}}, ::Pair{Symbol,Float64}, ::Pair{Symbol,Array{Int64,1}}, ::Vararg{Pair{Symbol,#s63} where #s63,N} where N) at /Users/forets/.julia/v0.6/Reachability/src/solve.jl:234
 [8] include_string(::String, ::String) at ./loading.jl:522
```

--- a/src/Utils/Utils.jl
+++ b/src/Utils/Utils.jl
@@ -303,45 +303,39 @@ by a `UnitRange{Int}`.
 ### Output
 
 New partition representation.
+
+### Notes
+
+We cannot mix `Int` and `UnitRange{Int}`, so we only create `Int` partitions if
+all blocks are 1D.
 """
 function convert_partition(partition::AbstractVector{<:AbstractVector{Int}})::Union{
         Vector{Int},
-        Vector{UnitRange{Int}},
-        Vector{Union{UnitRange{Int}, Int}}}
-    # are there 1D blocks and/or kD blocks?
-    has_1D_blocks = false
+        Vector{UnitRange{Int}}}
+    # are there kD blocks for k > 1?
     has_kD_blocks = false
     for (i, block) in enumerate(partition)
-        if length(block) == 1
-            has_1D_blocks = true
-            if has_kD_blocks
-                break
-            end
-        else
+        if length(block) > 1
             has_kD_blocks = true
-            if has_1D_blocks
-                break
-            end
+            break
         end
     end
 
     # use optimal partition type
-    if !has_1D_blocks
-        partition_out = Vector{UnitRange{Int}}(length(partition))
-    elseif !has_kD_blocks
+    if !has_kD_blocks
+        # only 1D blocks
         partition_out = Vector{Int}(length(partition))
-    else
-        partition_out = Vector{Union{UnitRange{Int}, Int}}(length(partition))
-    end
-
-    # convert old partition representation to new one
-    for (i, block) in enumerate(partition)
-        if length(block) == 1
+        for (i, block) in enumerate(partition)
             partition_out[i] = block[1]
-        else
+        end
+    else
+        # at least one kD block for k > 1
+        partition_out = Vector{UnitRange{Int}}(length(partition))
+        for (i, block) in enumerate(partition)
             partition_out[i] = block[1]:block[end]
         end
     end
+
     return partition_out
 end
 

--- a/test/Reachability/unit_solve.jl
+++ b/test/Reachability/unit_solve.jl
@@ -61,3 +61,11 @@ A = randn(5, 5); X0 = BallInf(ones(5), 0.1)
 system = ContinuousSystem(A, X0)
 options = Options(Dict(:T=>0.1, :partition=>[1:2, 3:4, [5]], :block_types=>Dict(HPolygon=>[1:2, 3:4], Interval=>[[5]]), :vars=>[1,3]))
 s = solve(system, options)
+
+# ===============================
+# System with an odd dimension
+# ===============================
+A = [1 2 1.; 0 0. 1; -2 1 4]
+X0 = BallInf(ones(3), 0.1)
+system = ContinuousSystem(A, X0)
+s = solve(system, :T=>0.1, :vars=>1:3, :partition=>[1:2, 3:3])

